tools/tools.go

```diff
@@ -314,6 +318,71 @@ func EnvmanRun(envstorePth, workDirPth string, cmd []string) (int, error) {
 	return command.RunCommandInDirAndReturnExitCode(workDirPth, "envman", args...)
 }
 
+// EnvmanRunWithTimeout ...
+func EnvmanRunWithTimeout(envstorePth, workDirPth string, cmdArgs []string, timeout time.Duration) (int, error) {
+	logLevel := log.GetLevel().String()
+	args := []string{"--loglevel", logLevel, "--path", envstorePth, "run"}
+	args = append(args, cmdArgs...)
+
+	command := command.NewWithStandardOuts("envman", args...).SetStdin(os.Stdin).SetDir(workDirPth)
+
+	// create a new process group for our process and its child processes
+	cmd := command.GetCmd()
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+	if err := cmd.Start(); err != nil {
+		return 1, err
+	}
+
+	var timer *time.Timer
+
+	// Setpgid: true creates a new process group for cmd and its subprocesses
+	// this way cmd will not belong to its parent process group,
+	// cmd will not be killed when you hit ^C in your terminal
+	// to fix this, we listen and handle Interrupt signal manually
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
```

Comment:
add test: at least 2 steps with timeout -> does it cause any issues that we do multiple `signal.Notify(c, os.Interrupt)`? It seems to be a global notify